adds new content section (#24)
ecoacoustics/website: content/resources/help-centre/software-tips/github.md

---
title: GitHub
---

## What is it?

[GitHub](https://github.com/) is a website and service which lets people store
their code in the cloud, with a generous free option that has all the features
you will need to get started. In simple terms, you can think of it as being like

> a DropBox for developers. But more than being just a place to store code, the

Reviewer: ```suggestion
a DropBox for anyone who writes code. But more than being just a place to store code, the
```

> value of GitHub lies in its version control capabilities, and open source

Reviewer: version control: in the scientific context allows for reproducibility and sovereignty of IP (you can see the history of an idea develop).

collaborative community. 

> 

Reviewer: github also has tight integrations with zenodo which will allow researchers to mint DOIs for the software outputs

## When to use it

Whether you are writing code by yourself, within a small team, or even as part
of a large collaboration, GitHub can help you to safely and securely manage your
projects, and work together with others from all over the world.

{{% hint info %}}
**Fun fact:**  

> The Open Ecoacoustics website is hosted as a [repository](https://github.com/ecoacoustics/website) on GitHub!

Reviewer: ```suggestion
This very website is hosted as a [repository](https://github.com/ecoacoustics/website) on GitHub!

Anyone can contribute!
```

{{% /hint %}}

## Getting started

If you've never used GitHub or Git version control before, then a great place to
start is the [Hello World](https://docs.github.com/en/get-started/quickstart/hello-world) quick start
guide provided by GitHub. By following this short tutorial, you will have learnt
a number of skills such as:

- How to make an open source repository (a place where you can store your project)
- How to make a branch (a type of isolated version of your project where you can
  make changes without affecting the main project)
- How to make changes to a file, and "commit" those changes to GitHub
- How to open and merge a pull request - which pulls the content from the branch

>   you created, into your main project!

Reviewer: also link to the software carpentry courses.

perhaps also suggest using GitHub Desktop as an easy way to get started
